sdk/iot/core/src/az_iot_core.c

```diff
@@ -8,74 +8,41 @@
 #include <az_result.h>
 #include <az_span.h>
 
+#include <az_retry_internal.h>
+
 #include <_az_cfg.h>
 
-AZ_NODISCARD az_result az_iot_get_status_from_uint32(uint32_t status_int, az_iot_status* status)
+AZ_NODISCARD int32_t az_iot_retry_calc_delay(
+    int32_t operation_msec,
+    int16_t attempt,
+    int32_t min_retry_delay_msec,
+    int32_t max_retry_delay_msec,
+    int32_t random_msec)
 {
-  switch (status_int)
+  _az_PRECONDITION_RANGE(0, operation_msec, INT32_MAX - 1);
```

Comment:
Why is `INT32_MAX` not allowed?

Reply:
It's allowed but it'll generate a compile-time overflow warning and break the build.